x/autopilot/keeper/redeem_stake.go

```diff
@@ -18,74 +17,64 @@ import (
 func (k Keeper) TryRedeemStake(
 	ctx sdk.Context,
 	packet channeltypes.Packet,
-	newData transfertypes.FungibleTokenPacketData,
-	packetMetadata types.StakeibcPacketMetadata,
+	transferPacketData transfertypes.FungibleTokenPacketData,
+	autopilotMetadata types.StakeibcPacketMetadata,
 ) error {
 	params := k.GetParams(ctx)
 	if !params.StakeibcActive {
 		return fmt.Errorf("packet forwarding param is not active")
 	}
 
-	// In this case, we can't process a liquid staking transaction, because we're dealing IBC tokens from other chains
-	if !transfertypes.ReceiverChainIsSource(packet.GetSourcePort(), packet.GetSourceChannel(), newData.Denom) {
-		return fmt.Errorf("the ibc tokens are not supported for redeem stake")
+	// At this point in the stack, the denom's in the packet data appear as they existed on the sender zone,
+	//   but as a denom trace instead of a hash
+	// Meaning, for native stTokens, the port and channel on the host zone are part of the denom
+	//   (e.g. transfer/{channel-on-hub}/stuatom)
+	// Only stride native stTokens can be redeemed, so we confirm that the denom's prefix matches
+	//   the packet's "source" channel (i.e. the channel on the host zone)
+	if !transfertypes.ReceiverChainIsSource(packet.GetSourcePort(), packet.GetSourceChannel(), transferPacketData.Denom) {
+		return fmt.Errorf("the ibc token %s is not supported for redeem stake", transferPacketData.Denom)
 	}
 
 	voucherPrefix := transfertypes.GetDenomPrefix(packet.GetSourcePort(), packet.GetSourceChannel())
-	stAssetDenom := newData.Denom[len(voucherPrefix):]
+	stAssetDenom := transferPacketData.Denom[len(voucherPrefix):]
 	if !stakeibctypes.IsStAssetDenom(stAssetDenom) {
 		return fmt.Errorf("not a liquid staking token")
 	}
 
 	hostZoneDenom := stakeibctypes.HostZoneDenomFromStAssetDenom(stAssetDenom)
 
-	amount, ok := sdk.NewIntFromString(newData.Amount)
-	if !ok {
+	amount, ok := sdk.NewIntFromString(transferPacketData.Amount)
+	if !ok || amount.IsNegative() {
```

Comment:
what's the scenario where `eval(ok is true and amount < 0) is true`?

Reply:
if amount was, say, "-1000", it would be successfully parsed into an int (i.e. `ok == true`), but it would be negative

Reply:
shouldn't the underlying ibc transfer module filter packets with negative amounts out? 

if it's just a sanity check, including it seems fine (it might make sense to check `amount.LTE(0)`)

Reply:
I don't think it matters much either way cause we validate that amount again in the redeem stake validate basic

I think the only reason I put it in was because it was the only meaningful check that was done in the coin.Validate function that I removed. 

I'll remove the isPositive in the autopilot-redeem branch